Save/book_storage.rb

```diff
@@ -0,0 +1,46 @@
+require 'json'
+require_relative './storage'
+require_relative './rental_storage'
+require './book'
+
+class BookStorage < Storage
+  @books = []
+
+  class << self
+    attr_accessor :books
+  end
+
+  def self.fetch
+    books = if File.exist?('./data/books.json')
+              JSON.parse(File.read('./data/books.json'))
+            else
+              []
+            end
+    deserialize(books)
+  end
+
+  def self.save(books)
+    books.each do |book|
+      BookStorage.books.push(serialize(book))
+    end
+
+    File.write('./data/books.json', JSON.pretty_generate(BookStorage.books))
+  end
```

Comment:
- [ ] When I cloned the repository, there was no directory called `data`. So when I created all books, persons and rentals during usage of the application, I got an error upon exiting the application. This was because you didn't have a code to ensure that the application doesn't break when the data directory is not present. To fix this, I would suggest you create the data directory unless the data directory exists. It would be like this:

`    Dir.mkdir('./data') unless Dir.exist?('./data')
`
#### Kindly use this guard clause in the `save` method of your `BookStorage`, `PeopleStorage`, and `RentalStorage` classes 👍 😎

Reply:
Fixed.